sycl/test/sub_group/attributes.cpp

```diff
@@ -0,0 +1,108 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//==------- attributes.cpp - SYCL sub_group attributes test ----*- C++ -*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "helper.hpp"
+#include <CL/sycl.hpp>
+
+#define KERNEL_FUNCTOR_WITH_SIZE(SIZE)                                         \
+  class KernelFunctor##SIZE {                                                  \
+  public:                                                                      \
+    [[cl::intel_reqd_sub_group_size(SIZE)]] void                               \
+    operator()(cl::sycl::nd_item<1> Item) {                                    \
+      const auto GID = Item.get_global_id();                                   \
+    }                                                                          \
+  };
+
+KERNEL_FUNCTOR_WITH_SIZE(1);
```

Comment:
Overall this test looks good. I have only one concern that it might not work properly on any OpenCL RT with sub-groups support.

The problem is that **all** kernel functions described here are going to be included into a **one** device program which will be compiled later by a device compiler - and this might cause errors because apparently device doesn't have to support all sub-group sizes listed in here and it is legal to fail compilation if requested sub-group size is not supported.

Probably it is a bug in our implementation, that `build_with_kernel_type` actually performs build of **all** kernels instead of just one, but I don't know SYCL spec so good to say for sure.

Reply:
I remembered there's a ticket created by Evgeniy describing the issue of building all kernels.